### ISSUE TITLE
Cannot read property 'handle' of undefined

### ISSUE THREAD

**Issue by Remo** [nuxt/bridge#314]
### Environment

Operating System: Darwin
Node Version: v16.11.0
Nuxt Version: 2.16.0-27358576.777a4b7f
Package Manager: yarn@1.22.17
Bundler: Webpack
User Config: target, head, css, plugins, components, buildModules, modules, vuetify, build
Runtime Modules: -
Build Modules: @nuxtjs/vuetify@1.12.3

### Reproduction

https://github.com/timgoeller/nuxt-bug
https://github.com/Remo/laravel-nuxt-bridge

### Describe the bug

This bug has already been reported twice, the issues are closed though and since I can't re-open them, I'm posting it again hoping to get some help. Here are the two issues:
https://github.com/nuxt/bridge/issues/138
https://github.com/nuxt/bridge/issues/227

In short, calling `nuxi dev` results in this error:

```
 ERROR  Cannot read properties of undefined (reading 'callHook')                                                                           16:23:36

  at Object.invoke (node_modules/nuxi/dist/chunks/dev.mjs:6773:31)
  at runMicrotasks (<anonymous>)
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
  at async _main (node_modules/nuxi/dist/chunks/index.mjs:384:7)
```

**Comment by Luferov** [comment]
May, you need start `dev:prepare` and then `dev`?

**Comment by Remo** [comment]
@Luferov according to this, there's no `dev:prepare`: https://v3.nuxtjs.org/bridge/overview. However, I went ahead and copied it from here https://github.com/nuxt/bridge/blob/main/package.json
When I ran the command I simply got a `playground` directory with a nuxt project, but since I already have one where I'd like to integrate nuxt bridge, this doesn't help.

**Comment by Luferov** [comment]
> @Luferov according to this, there's no `dev:prepare`: https://v3.nuxtjs.org/bridge/overview. However, I went ahead and copied it from here https://github.com/nuxt/bridge/blob/main/package.json
> When I ran the command I simply got a `playground` directory with a nuxt project, but since I already have one where I'd like to integrate nuxt bridge, this doesn't help.

Oops, I thinks it’s about nuxt-bridge project and playground inside 😯

**Comment by jonathanmach** [comment]
I'm facing the same issue, did you make any progress @Remo ?

**Comment by Remo** [comment]
@jonathanmach I'm afraid, I'm still stuck

**Comment by jonathanmach** [comment]
I managed to solve it by wrapping the nuxt.config object inside `defineNuxtConfig()`:

![image](https://user-images.githubusercontent.com/12788052/172560270-2b8373a7-a5bf-4a8a-aa79-32d2bb4c3b9d.png)


Source: https://v3.nuxtjs.org/bridge/overview#update-nuxtconfig

**Comment by Remo** [comment]
I already have that https://github.com/Remo/laravel-nuxt-bridge/blob/master/client/nuxt.config.js

**Comment by danielroe** [comment]
This issue is caused by running `nuxi dev` on a project that doesn't have a config defined with `defineNuxtConfig({})`.

You've mentioned two repositories:
* [timgoeller/nuxt-bug](https://github.com/timgoeller/nuxt-bug) - this just exports a bare object
* [Remo/laravel-nuxt-bridge](https://github.com/Remo/laravel-nuxt-bridge) - this is more complex, but basically your nuxt.config is within a directory and `nuxi` doesn't know to look there for the config. Changing your command from `nuxi dev` to `nuxi dev client` resolves the issue.

**Comment by Remo** [comment]
@danielroe thanks, so simple!

**Comment by dmitriy-grablyov** [comment]
Hello!

I have properly defined `defineNuxtConfig` with the following bridge section

```
bridge: {
    vite: true,
    meta: false,
    nitro: true,
    app: true,
    capi: false,
    transpile: true,
    scriptSetup: false,
    imports: false,
    constraints: false,
    typescript: false,
```

And I am still getting `Cannot read properties of undefined (reading 'callHook')` , irregardless whether I call `npx nuxi dev` or `npx nuxi build && npx nuxi preview`

I can assure that it is using the bridge configuration, because well it indeed uses vite to build sources as defined in nuxt.config.js

I tried to trace an error and only was able to trace it to this line https://github.com/nuxt/bridge/blob/1313524c5c84421837263d45202a64229881f864/packages/bridge/src/runtime/app.ts#L39
In runtime `ctx.$_nuxtApp` evaluates to undefined and I tried hard, but was not able to find the place in the sources where  `ctx.$_nuxtApp` is getting actually assigned to context. Where does this magic happen?

Thanks in advance!
